scipy/ndimage/_filters.py

```diff
@@ -1409,6 +1411,33 @@ def maximum_filter(input, size=None, footprint=None, output=None,
                               cval, origin, 0, axes)
 
 
+def hampel(input: np.ndarray, win_size: int, thresh=1):
```

Comment:
You cannot expect a public function to only accept numpy arrays anymore. The way to go is to accept whatever array-like input the user supplies, then
- call `np.asarray` on the array-likes, and
- add a matching `hampel_signature` function to https://github.com/scipy/scipy/blob/main/scipy/ndimage/_delegators.py; this will get picked up by the decorator from `_support_alternative_backends.py`

I should add that you're the first person to add things to ndimage after the array api change, so you're test-driving the process (sorry).